### ISSUE TITLE
Second and further calls of Image#toDataURL return a transparent image data URL

### ISSUE THREAD

**Issue by finom** [kangax/fabric.js#4641]
## Version
2.0.0-rc.4

## Test Case
http://jsfiddle.net/Da7SP/810/

## Steps to reproduce

Open the test case.

## Expected Behavior

All boxes with red border should contain the original image. In other words, `Image#toDataURL` should always return the same result.

## Actual Behavior

Calling a custom function `extractImage` from the test case multiple times should append correct images to `document.body`. Instead, it appends the original image after the first call only but other calls append a transparent image of original size. 

In other words, calling `Image#toDataURL` twice will return different results.

![image](https://user-images.githubusercontent.com/1082083/35221073-7ada86c8-ff81-11e7-98a5-583124a7a2a3.png)

Also look at the console:

![image](https://user-images.githubusercontent.com/1082083/35221099-9285ac62-ff81-11e7-9a9b-a3b4cdad6a67.png)

The first call is different from others.

**Comment by asturur** [comment]
![image](https://user-images.githubusercontent.com/1194048/35432310-eab0706a-027f-11e8-8aa8-9a76ac87219b.png)

what a sneaky bug!
Found the problem i m fixing it.

**Comment by finom** [comment]
@asturur great! When it's going to be published?

**Comment by asturur** [comment]
is on the master branch. if i manage to make a release in the weekend,
saturday or sunday.

2018-01-26 17:08 GMT+01:00 Andrey Gubanov <notifications@github.com>:

**Comment by finom** [comment]
@asturur good, thank you.

Why not to move to automatic releases via semantic-release btw?

**Comment by asturur** [comment]
Because our automated test suite is not good enough.
every 10 releases 9 would be broken. while when i do manually i check all the commits, i try to see if i missed something.

**Comment by finom** [comment]
Got it. Can't wait to get the fix! Before that I'm going to build fabricjs locally 😐

**Comment by finom** [comment]
@asturur made my own build and the issue listed at the issue header is fixed only partially. It doesn't return the transparent image every time but it returns it in some other cases (which look random for me for now).  Trying to reproduce at jsfiddle.

**Comment by asturur** [comment]
mmm i do not think v1 is disposing images at all.
Are you sure?

**Comment by finom** [comment]
@asturur couldn't reproduce it on JSFiddle, so I cannot be 100% sure yet. My ideas are:
- This is incorrect library usage (i.e. my issue)
- This is CORS issue (images are hosted on external host)
- This is a library issue
